### ISSUE TITLE
Various false negatives with BigQuery sql

### ISSUE THREAD

**Issue by adamcunnington-mlg** [sqlfluff/sqlfluff#4017]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

**Given the following configuration:**
```yaml
[sqlfluff]
dialect = bigquery
encoding = utf-8
exclude_rules = L025,L051
processes = 0
templater = dbt

[sqlfluff:indentation]
indented_on_contents = False

[sqlfluff:templater:dbt]
profile = mlg-apollo-data-prod

[sqlfluff:rules]
max_line_length = 120
single_table_references = unqualified
tab_space_size = 2

[sqlfluff:rules:L010]
capitalisation_policy = upper

[sqlfluff:rules:L042]
forbid_subquery_in = both

[sqlfluff:rules:L059]
prefer_quoted_identifiers = True

[sqlfluff:rules:L064]
preferred_quoted_literal_style = double_quotes
```

**when linting the following file:**
```sql
SELECT
  `date`,
  FORMAT_DATE("%A", `date`) AS `day`,
  FORMAT_DATE("%u", `date`) AS `day_of_week`,
  FORMAT_DATE("%B", `date`) AS `month`,
  EXTRACT(DAY FROM `date`) AS `day_of_month`,
  EXTRACT(DAYOFYEAR FROM `date`) AS `day_of_year`,
  EXTRACT(WEEK(MONDAY) FROM `date`) AS `week_number`,
  EXTRACT(ISOWEEK FROM `date`) AS `iso_week_number`,
  EXTRACT(MONTH FROM `date`) AS `month_of_year`,
  EXTRACT(QUARTER FROM `date`) AS `quarter`,
  EXTRACT(YEAR FROM `date`) AS `year`,
  EXTRACT(ISOYEAR FROM `date`) AS `iso_year`,
  DATE_TRUNC(`date`, `ISOWEEK`) AS `iso_week_commencing_date`
FROM UNNEST(GENERATE_DATE_ARRAY({{ var("infinity_past") }}, "2099-12-31")) AS `date`
```

**I receive the following output which I believe to all be false negatives:**

```
== [models/meta/meta_dates.sql] FAIL
L:   1 | P:   1 | L034 | Select wildcards then simple targets before calculations
                       | and aggregates.
L:   8 | P:   3 | L059 | Missing quoted identifier EXTRACT.
L:   8 | P:  10 |  PRS | Line 8, Position 10: Found unparsable section:
                       | '(WEEK(MONDAY) FROM `date`) AS `week_numb...'
WARNING: Parsing errors found and dialect is set to 'bigquery'. Have you configured your dialect correctly?
```
Notably:
* There are no targets after calculations/aggregates.
* EXTRACT is a function and should not be quoted. I tried quoting the whole function (invalid syntax: **`EXTRACT(DAY FROM date)`**) and just the day part identifier (valid but doesn't get rid of lint message: **EXTRACT(`DAY` FROM `date`)**.
* Unparseable section suggests the linter is not aware of this part of the BigQuery SQL dialect?

### Expected Behaviour

I expected no lint issues with this SQL.

### Observed Behaviour

As described above.

### How to reproduce

Clear from above.

### Dialect

Clear from configuration.

### Version

sqlfluff, version 1.2.1

### Configuration

As above.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by barrywhart** [comment]
Can you please create a separate issue for each problem you found? That will make it easier for the maintainers to address them. Thanks!

**Comment by yoichi** [comment]
The unparseable section error was fixed in https://github.com/sqlfluff/sqlfluff/pull/3787 (included in sqlfluff version 1.3.1).

**Comment by adamcunnington-mlg** [comment]
Actually, all 3 issues seem to have gone with 1.4.1 so I am closing this ticket - sorry and thanks!
